### PR TITLE
EOS-15766: Design and implement the logic of resume after reboot

### PR DESCRIPTION
**Pull request by SomwanshiAnjali**
Signed-off-by: Anjali Somwanshi <anjali.somwanshi@seagate.com>

PR includes below changes for EOS-15766:

1. Create service to resume after reboot in between software update process
2. Create script to run provisioner commands by checking pillar data 
for now, below format considered for pillar:

```
update_states:
  pre_boot:
    - state1
    - state2
  post_boot:
    - state1
    - state2
```
For now, considered these states as provisioner commands

### CODE REVIEW

**Review by andkononykhin2 — APPROVED** [review]

File: cli/src/update-post-reboot.sh [hunk 1]
```diff
@@ -0,0 +1,67 @@
+#!/bin/bash
```
Comment (ypise) [inline cli/src/update-post-reboot.sh:1]: @SomwanshiAnjali 
The general instructions for us are to have lesser batch scripts and more Python code.
Is it possible to convert this to Python code?
Also, move this code under dedicated upgrade directory:
`cli/src/upgrade/update-post-reboot.sh`

**Review by ypise — APPROVED** [review]

File: cli/src/update-post-reboot.sh (hunk 1 above)
Comment (ypise) [inline cli/src/update-post-reboot.sh:1]: @SomwanshiAnjali
The general instructions for us are to have lesser batch scripts and more Python code.
Is it possible to convert this to Python code?
Also, move this code under dedicated upgrade directory:
`cli/src/upgrade/update-post-reboot.sh`

**Review by ypise — CHANGES_REQUESTED** [review]

File: cli/src/update-post-reboot.sh (hunk 1 above)
Comment (andkononykhin2) [reply cli/src/update-post-reboot.sh:1]: @ypise It's good to move that file to a separate dir as you suggested but I think we can keep bash for now. Actually it was me who asked @SomwanshiAnjali to do just a PoC without any restrictions regarding implementation. I believe python will definitely come into play soon in scope of the next upgrade PR from Anjali.

**Review by ypise — COMMENTED** [review]
Forgot to update this one.

File: cli/src/update-post-reboot.sh (hunk 1 above)
Comment (SomwanshiAnjali) [reply cli/src/update-post-reboot.sh]: Addressed

File: cli/src/update-post-reboot.sh [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+    post_boot_op=$(salt-call pillar.get update_states:post_boot --output=newline_values_only)
+    if [[ ${post_boot_op} ]]; then
+        for state in ${post_boot_op}; do
+            echo "Performing ${state} on target nodes" 2>&1 | tee -a "${LOG_FILE}"
+            provisioner "${state}"
+        done 
+    else
+        echo "No operations to perform !!" 2>&1 | tee -a "${LOG_FILE}""
+    fi
```
[the remark this answers is not in the document]
Comment (SomwanshiAnjali) [reply cli/src/update-post-reboot.sh]: Addressed

**Review by ypise — APPROVED** [review]

**Review by ypise — APPROVED** [review]

**Pull request merged**